libs/ui/src/components/molecules/GatedInput/GatedInput.tsx

```diff
@@ -0,0 +1,50 @@
+import { ReactNode, useEffect, useState, FunctionComponent } from 'react';
+import { FieldValues, useFormContext } from 'react-hook-form';
+import { CheckedState } from '@radix-ui/react-checkbox';
+import { FieldWrapper } from '../FieldWrapper';
+import { Checkbox } from '../../atoms/Checkbox';
+import { Buildable, IGatedInput } from '../../../types/formAndField';
+import { GatedInputBox } from './GatedInput.styles';
+
+type GatedInputProps = IGatedInput & {
+  gatedId: string;
+  gateLabel: string;
+  input: FunctionComponent<Buildable<IGatedInput>>;
+  updateHelperMsg?: (value: FieldValues) => string; 
```

Comment:
What is the use case for passing in a callback for generating helper text?

Reply:
The Expiration Date component https://github.com/HausDAO/daohaus-monorepo/issues/498 requires updating the helper text based on the input values. I could set this text as children but I'm wondering where exactly this explanation text updates should be handled later in the form

<img width="532" alt="image" src="https://user-images.githubusercontent.com/926341/184912196-3de75a8a-8adf-410a-87de-1bdc4db486be.png">